test_unstructured/partition/common/test_metadata.py

```diff
@@ -62,73 +62,139 @@ def but_it_returns_None_when_there_is_no_file_at_that_path(self, tmp_path: pathl
-        ListItem(text="ListItem", metadata=ElementMetadata(category_depth=1)),  # 5
-        ListItem(text="ListItem"),  # 6
-        CheckBox(element_id="some-id-1", checked=True),  # 7
-        Title(text="Title 2"),  # 8
-        ListItem(text="ListItem"),  # 9
-        ListItem(text="ListItem"),  # 10
-        Text(text="Text"),  # 11
-    ]
-    elements = set_element_hierarchy(elements_to_set)
-
-    assert (
-        elements[1].metadata.parent_id == elements[0].id
-    ), "NarrativeText should be child of Title"
-    assert (
-        elements[2].metadata.parent_id == elements[0].id
-    ), "FigureCaption should be child of Title"
-    assert elements[3].metadata.parent_id == elements[0].id, "ListItem should be child of Title"
-    assert elements[4].metadata.parent_id == elements[3].id, "ListItem should be child of Title"
-    assert elements[5].metadata.parent_id == elements[3].id, "ListItem should be child of Title"
-    assert elements[6].metadata.parent_id == elements[0].id, "ListItem should be child of Title"
-    # NOTE(Hubert): moving the category field to Element, caused this to fail.
-    # Checkboxes will soon be deprecated, then we can remove the test.
-    # assert (
-    #         elements[7].metadata.parent_id is None
-    # ), "CheckBox should be None, as it's not a Text based element"
-    assert elements[8].metadata.parent_id is None, "Title 2 should be child of None"
-    assert elements[9].metadata.parent_id == elements[8].id, "ListItem should be child of Title 2"
-    assert elements[10].metadata.parent_id == elements[8].id, "ListItem should be child of Title 2"
-    assert elements[11].metadata.parent_id == elements[8].id, "Text should be child of Title 2"
-
-
-def test_set_element_hierarchy_custom_rule_set():
-    elements_to_set = [
-        Header(text="Header"),  # 0
-        Title(text="Title"),  # 1
-        NarrativeText(text="NarrativeText"),  # 2
-        Text(text="Text"),  # 3
-        Title(text="Title 2"),  # 4
-        FigureCaption(text="FigureCaption"),  # 5
-    ]
+class Describe_set_element_hierarchy:
+
+    def it_applies_default_ruleset(self):
+        elements = [
+            Title(element_id="0", text="Title0"),
+            Text(element_id="1", text="Text0"),
+            Header(element_id="2", text="Header0"),
+            Text(element_id="3", text="Text1"),
+            Title(element_id="4", text="Title1"),
+            Text(element_id="5", text="Text2"),
+        ]
+
+        result = set_element_hierarchy(elements)
+
+        assert result[0].metadata.parent_id is None
+        assert result[1].metadata.parent_id == "0"  # Text0 is under Title0
+        assert result[2].metadata.parent_id is None  # Header0 is higher than Title0
+        assert result[3].metadata.parent_id == "2"  # Text1 is under Header0
+        assert result[4].metadata.parent_id == "2"  # Title1 is under Header0
+        assert result[5].metadata.parent_id == "4"  # Text2 is under Title1, which is under Header0
```

Comment:
Hmm, interesting. So we don't actually need a `.metadata.category_level` value for this to work? I'm assuming that's only necessary when the title or whatever is not at the top level; that makes sense now that I think of it.

The clarity of this test is good; makes details like that pop out :)